AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -428,11 +430,33 @@ - (void)setAsyncDataSource:(id<ASCollectionDataSource>)asyncDataSource
     _asyncDataSourceFlags.collectionNodeNodeForSupplementaryElement = [_asyncDataSource respondsToSelector:@selector(collectionNode:nodeForSupplementaryElementOfKind:atIndexPath:)];
 
 
-    ASDisplayNodeAssert(_asyncDataSourceFlags.collectionNodeNumberOfItemsInSection || _asyncDataSourceFlags.collectionViewNumberOfItemsInSection, @"Data source must implement collectionNode:numberOfItemsInSection:");
-    ASDisplayNodeAssert(_asyncDataSourceFlags.collectionNodeNodeBlockForItem
-                        || _asyncDataSourceFlags.collectionNodeNodeForItem
-                        || _asyncDataSourceFlags.collectionViewNodeBlockForItem
-                        || _asyncDataSourceFlags.collectionViewNodeForItem, @"Data source must implement collectionNode:nodeBlockForItemAtIndexPath: or collectionNode:nodeForItemAtIndexPath:");
+    // Make assertions about what they implement, unless they are suppressed.
+    if (self.test_suppressCallbackImplementationAssertions == NO) {
+      // If they implement the functional method, they should not implement the classic ones.
+      if (_asyncDataSourceFlags.dataForCollectionNode) {
+        SEL legacySelectors[] = {
```

Comment:
This is cute, but I'm not sure we should call it legacy yet :).  Maybe imperative?  If you want to separate out the deprecated ones, those are legacy at this point :)